test/utils/wasm_engine.hpp

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include "types.hpp"
+#include <memory>
+#include <optional>
+
+namespace fizzy
+{
+struct execution_result2
```

Comment:
Maybe this should be `execution_result` in another (nested?) namespace?

Reply:
Number of options here, in order of my preferences:
1. Use `std::optional` in the original `execute_result`.
2. Use original `execute_result` here.
3. Rename to something else. Can be `fizzy::test::execution_result` but I expect it might be confusing to have `fizzy::execution_result` and `fizzy::test::execution_result`.

Still, adding all test utils, including this one into `fizzy::test` sound like good idea.

Reply:
Why not nest it inside `WasmEngine`? `WasmEngine::Result` ?